tests/unit/src/Internal/Payment/State/ProcessedStateTest.php

```diff
@@ -93,8 +101,24 @@ public function test_complete_processing_will_transition_to_completed_state() {
 			->with( CompletedState::class, $this->mock_context )
 			->willReturn( $mock_completed_state );
 
+		$this->mock_legacy_proxy
+			->expects( $this->once() )
+			->method( 'call_function' )
+			->with( 'wc' )
+			->willReturnCallback(
+				function () {
+					$mock_cart = $this->getMockBuilder( \stdClass::class )
+						->setMethods( [ 'empty_cart' ] )
+						->getMock();
+					return (object) [
+						'cart' => $mock_cart,
```

Comment:
Shouldn't we verify that the cart is actually cleared?

Reply:
Wouldn't that be testing whether the WC `empty_cart` does its job, rather than a unit test?

Reply:
Similarly to the other conversation (https://github.com/Automattic/woocommerce-payments/pull/7737#discussion_r1401670362), we just want to make sure that `empty_cart` is called. Currently, you are preparing it to be mocked (`setMethods`) but never set any expectations for it to be called.

Reply:
Test added.